src/table_builder.cc

```diff
@@ -87,35 +116,43 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
       // doing level merge.
       if (get_status.ok()) {
         std::string index_value;
-        AddBlob(ikey.user_key, record.value, &index_value);
+        BlobRecord blob_record;
+        blob_record.key = ikey.user_key;
+        blob_record.value = record.value;
+        BlobFileBuilder::OutContexts contexts;
+        AddBlob(blob_record, ikey, &contexts);
+        AddToBaseTable(contexts);
```

Comment:
Since we call `AddToBaseTable` every time after `AddBlob`, let's move the `AddToBaseTable` call to inside `AddBlob`.